### PR TITLE
fix: strip js comments

### LINKED ISSUE #45127: Basic JavaScript Curriculum - code comments are evaluated as part of the solution - multiple exercises affected
## Affected pages
- https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-javascript/explore-differences-between-the-var-and-let-keywords
- https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-javascript/declare-a-read-only-variable-with-the-const-keyword
- https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-javascript/increment-a-number-with-javascript

## Your code

```
// 009 Explore Differences Between the var and let Keywords.js

/*
  unlike var, when you use let, a variable with the 
  same name can only be declared once.

  Update the code so it only uses the let keyword.
*/

let catName = "Oliver";
let catSound = "Meow!";
```
```
// 015 Increment a Number with JavaScript.js

/*
  Change the code to use the ++ operator on myVar.
*/

let myVar = 87;

// Only change code below this line
//myVar = myVar + 1;

myVar++

console.log(myVar);

```

## Expected behavior
Validate as correct

## Screenshots
image

## System

- Device: [e.g. iPhone6, Laptop] Desktop
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.4] Manjaro Gnome
- Browser: [e.g. chrome, safari] Firefox
- Version: [e.g. 22] 97.0

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Checklist:

- [ ] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [ ] My pull request targets the `main` branch of freeCodeCamp.
- [ ] I have tested these changes either locally on my machine, or GitPod.

Closes #45127

I'm not sure why the `removeJSComments` helper was not being made available, but this PR enables it + fixes the tests in the linked issue.

### CODE REVIEW

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/declare-a-read-only-variable-with-the-const-keyword.md [hunk 1]
```diff
@@ -34,15 +34,15 @@ Change the code so that all variables are declared using `let` or `const`. Use `
 `var` should not exist in your code.
 
 ```js
-(getUserInput) => assert(!getUserInput('index').match(/var/g));
+(getUserInput) => assert(!__helpers.removeJSComments(getUserInput('index')).match(/var/g));
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/declare-a-read-only-variable-with-the-const-keyword.md:37]:
```suggestion
assert.notMatch(__file('script.js'), /var/);
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/declare-a-read-only-variable-with-the-const-keyword.md [hunk 2]
```diff
@@ -34,15 +34,15 @@ Change the code so that all variables are declared using `let` or `const`. Use `
...
 ```
 
 You should change `fCC` to all uppercase.
 
 ```js
 (getUserInput) => {
   assert(getUserInput('index').match(/(FCC)/));
-  assert(!getUserInput('index').match(/fCC/));
+  assert(!__helpers.removeJSComments(getUserInput('index')).match(/fCC/));
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/declare-a-read-only-variable-with-the-const-keyword.md:45]:
```suggestion
  assert.notMatch(__file('script.js'), /fCC/);
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Something like this should work more better.

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍 

Are we just closing the one issue?

---



Just so we are on the same page:
1) The global flag is redundant, and actually just a performance hit
2) There is not specific need here for the named assertions - I just happen to be in the habit of trying to use them.

**Comment by nhcarrigan** [comment]
Do we want to close the other issue? This only fixes these challenges, not the underlying cause.

**Review by raisedadead — APPROVED** [review]

**Pull request merged**